test/helper/Testcafe_test.js

```diff
@@ -0,0 +1,701 @@
+const TestHelper = require('../support/TestHelper');
+const Testcafe = require('../../lib/helper/Testcafe');
+const expect = require('chai').expect;
+const assert = require('assert');
+const path = require('path');
+const AssertionFailedError = require('../../lib/assert/error');
+const formContents = require('../../lib/utils').test.submittedData(path.join(__dirname, '/../data/app/db'));
+const expectError = require('../../lib/utils').test.expectError;
+const webApiTests = require('./webapi');
+const FileSystem = require('../../lib/helper/FileSystem');
+
+let I;
+let browser;
+let page;
+let FS;
+const siteUrl = TestHelper.siteUrl();
+
+describe('Testcafe', function () {
+  this.timeout(35000);
+  this.retries(1);
+
+  before(() => {
+    global.codecept_dir = path.join(__dirname, '/../data');
+
+    I = new Testcafe({
+      url: siteUrl,
+      windowSize: '500x700',
+      show: true,
+      waitForTimeout: 5000,
+      chrome: {
+        args: ['--no-sandbox', '--disable-setuid-sandbox'],
+      },
+    });
+    I._init();
+    return I._beforeSuite();
+  });
+
+  beforeEach(() => {
+    webApiTests.init({
+      I, siteUrl,
+    });
+    return I._before().then(() => {
+      page = I.page;
+      browser = I.browser;
+    });
+  });
+
+  afterEach(() => {
+    return I._after();
+  });
+
+  describe('open page : #amOnPage', () => {
+    it('should open main page of configured site', async () => {
+      await I.amOnPage('/');
+      const url = await I.grabCurrentUrl();
+      await url.should.eql(`${siteUrl}/`);
+    });
+    it('should open any page of configured site', async () => {
+      await I.amOnPage('/info');
+      const url = await I.grabCurrentUrl();
+      return url.should.eql(`${siteUrl}/info`);
+    });
+
+    it('should open absolute url', async () => {
+      await I.amOnPage(siteUrl);
+      const url = await I.grabCurrentUrl();
+      return url.should.eql(`${siteUrl}/`);
+    });
+  });
+
+  // describe('grabDataFromPerformanceTiming', () => {
+  //   it('should return data from performance timing', async () => {
+  //     await I.amOnPage('/');
+  //     const res = await I.grabDataFromPerformanceTiming();
+  //     expect(res).to.have.property('responseEnd');
+  //     expect(res).to.have.property('domInteractive');
+  //     expect(res).to.have.property('domContentLoadedEventEnd');
+  //     expect(res).to.have.property('loadEventEnd');
+  //   });
+  // });
+
+  // webApiTests.tests();
```

Comment:
The most important is to make those tests pass as they are the same for all helpers.
Other tests are helper specific.

Reply:
Will focus on that